core/src/components/progress-bar/progress-bar.tsx

```diff
@@ -0,0 +1,84 @@
+import { Component, ComponentInterface, Prop } from '@stencil/core';
+
+import { Color, Mode } from '../../interface';
+import { createColorClasses } from '../../utils/theme';
+
+@Component({
+  tag: 'ion-progress-bar',
+  styleUrls: {
+    ios: 'progress-bar.ios.scss',
+    md: 'progress-bar.md.scss'
+  },
+  shadow: true
+})
+export class ProgressBar implements ComponentInterface {
+
+  /**
+   * The mode determines which platform styles to use.
+   */
+  @Prop() mode!: Mode;
+
+  /**
+   * Style of the progress bar
+   * Options are `"determinate"` (no animation), `"indeterminate"` (animate from left to right),
+   * and `"buffer"` (shows circle points)
+   */
+  @Prop() type: 'determinate' | 'indeterminate' | 'buffer' = 'determinate';
+
+  /**
+   * Reverse the progress bar
+   */
+  @Prop() reversed = false;
+
+  /**
+   * Only on type `"determinate"` and  `"buffer"`: The width of the progress bar in percent - 0 ... 100
+   */
+  @Prop() value = 0;
+
+  /**
+   * Only on type `"buffer"`: The width of the buffer in percent - 0 ... 100
+   */
+  @Prop() buffer = 0;
+
+  /**
+   * The color to use from your application's color palette.
+   * Default options are: `"primary"`, `"secondary"`, `"tertiary"`, `"success"`, `"warning"`, `"danger"`, `"light"`, `"medium"`, and `"dark"`.
+   * For more information on colors, see [theming](/docs/theming/basics).
+   */
+  @Prop() color?: Color;
+
+  hostData() {
+    const { color, type, reversed } = this;
+    return {
+      class: {
+        ...createColorClasses(color),
+        [`progress-bar-${type}`]: true,
+        'progress-bar-reversed': reversed,
+      }
+    };
+  }
+
+  render() {
+    const content = [];
+    if (this.type === 'indeterminate') {
+      content.push(
+        <div class="indeterminate-bar-primary"><span class="progress-indeterminate"></span></div>,
+        <div class="indeterminate-bar-secondary"><span class="progress-indeterminate"></span></div>
+      );
+    } else {
+      const percent = this.value / 100;
```

Comment:
- 0 to 1, feels more "standard" to me, arguments against it?
- we should clamp the value to the valid range [0, 1] or [0, 100]
https://github.com/ionic-team/ionic/blob/master/core/src/utils/helpers.ts#L47

Reply:
I also thought about it, but I think 0...100 is better (angular material is also using this way) and it feels more natural https://material.angular.io/components/progress-bar/overview

Reply:
Looks like Apple uses [0, 1] in UIProgressView: https://developer.apple.com/documentation/uikit/uiprogressview/1619846-setprogress
Material Web Components uses [0, 1]: https://material.io/develop/web/components/linear-progress/
Angular uses [0, 100] https://developer.android.com/reference/android/widget/ProgressBar

Reply:
Clamp is now integrated for value and buffer